docs/design-guides/architecture-overview.md

```diff
@@ -0,0 +1,49 @@
+> Note: You can verify the configuration through app registration used for user sign in flow for Azure Active Directory instance. Go to the specific app registration within Azure Active Directory through [Azure Portal](https://portal.azure.com/) and check the Authentication tab to verify the tenancy configuration.
+- There are different Active Directory instances configured for different environments. 
+> Note: The sample only supports identity mapping for single environment within Graph Open Extensions used for storage of identity mapping.
+
+## Components
+![Diagram](../images/ACS-Authentication-Server-Sample_Overview-Flow.png)
+
+As seen from the Overview Diagram, the key components of the solution are:
+1. [Secure Web Api backed by Azure Active Directory](./secured-web-api-design.md)
+2. [Identity Mapping leveraging Graph Open Extensions](./Identity-Mapping-Design_Graph-Open-Extensions.md)
+3. Azure Communication Identity service which generates ACS Identity and Tokens. The interaction with ACS Identity service happens within [ACS sdk](https://docs.microsoft.com/en-us/azure/communication-services/concepts/sdk-options#sdks) used within the solution. 
+> Note: The `api/token/teams` endpoint does not leverage #2, as the M365 Azure Active Directory Identity is mapped to user's Teams Identity within ACS Identity Service.
+
+### Motivation for leveraging Graph Open Extensions for Identity Mapping
+The primary motivation comes from the fact that the Azure Communication Identity for the user could be co-located with rest of the information for the Active Active Directory user. And as that would also prevent the usage of additional storage for storing ACS Identity mappings corresponding to Azure Active Directory user Id.
+
+## Limitations
+- The open extensions on Azure Active Directory can only be supported upto 2 per user instance as suggested [here](https://docs.microsoft.com/en-us/graph/extensibility-overview#open-extension-limits). Hence supporting multi tenancy could be a challenge as different extensions might be desired for different tenants.
+- There is Graph open extensions rate limit of 455 requests per 10 seconds as suggested [here](https://docs.microsoft.com/en-us/graph/throttling#open-and-schema-extensions-service-limits)
+- The sample solution leveraging Graph open extensions might not be suitable for the scenarios where the Azure Active Directory instance is used for 3rd party application sign in as with [delegated permissions granted over Graph Api](https://docs.microsoft.com/en-us/graph/auth/auth-concepts#delegated-and-application-permissions). The 3rd Party application in this case with right delegated permissions would be able to access the open extension data of the user.
+> Note: From data privacy perspective, the ACS Identity is categorized as [EUPI data](https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/properties/devenv-dataclassification-property#property-value)
```

Comment:
Does this need to be here?

Reply:
From Architecture guidance perspective we need to let users know how they might want to make sure all the privacy compliance is met. May be there be Compliance sections in the doc.

Reply:
Actually this does not need to be here as already added on data privacy in guidance section. I will remove this part.